### PR TITLE
Injecting the object mapper in the flux client

### PR DESCRIPTION
**Pull request by aakn**
updated UT's (and added constructors for the same)

### CODE REVIEW

**Comment by aakn** [comment]
Tested this locally with SFS

File: client/src/main/java/com/flipkart/flux/client/runtime/FluxRuntimeConnectorHttpImpl.java [hunk 1]
```diff
@@ -50,9 +53,14 @@
     private final String fluxEndpoint;
     private final ObjectMapper objectMapper;
 
+    @VisibleForTesting
```
Comment (shyamakirala) [inline client/src/main/java/com/flipkart/flux/client/runtime/FluxRuntimeConnectorHttpImpl.java]: Same as above. Is it needed ?

File: client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java [hunk 2]
```diff
@@ -50,16 +52,18 @@
     @Inject
     private Provider<FluxRuntimeConnector> connectorProvider;
 
-    private final ObjectMapper objectMapper;
+    @Inject
+    private Provider<ObjectMapper> objectMapper;
```
Comment (shyamakirala) [inline client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java]: objectMapperProvider ? :)

File: client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java [hunk 3]
```diff
@@ -50,16 +52,18 @@
...
 
     public WorkflowInterceptor() {
-        this.objectMapper = new ObjectMapper();
     }
 
-    public WorkflowInterceptor(LocalContext localContext, Provider<FluxRuntimeConnector> connectorProvider) {
+    @VisibleForTesting
```
Comment (aakn) [inline client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java]: this is used to specify that a public method is only publicly available for testing. Should not be used in prod flows.

File: client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java (hunk 2 above)
Comment (aakn) [inline client/src/main/java/com/flipkart/flux/client/intercept/WorkflowInterceptor.java]: done

**Pull request merged**